### PR TITLE
Added ForceHttpsRedirectUris option applied to the RedirectUri and PostLogoutRedirectUri. Minor comment fixes.

### PR DESCRIPTION
**Pull request by pmaytak**
Added `ForceHttpsRedirectUris` config option. When set to true, forces the `ProtocolMessage.RedirectUri` and `ProtocolMessage.PostLogoutRedirectUri` in `OnRedirectToIdentityProvider` and `OnRedirectToIdentityProviderForSignOut` event handlers, respectively, to use the HTTPS scheme.

Plus minor comment fixes.

Related issues: #175, #115.

### CODE REVIEW

File: src/Microsoft.Identity.Web/MicrosoftIdentityOptions.cs [hunk 1]
```diff
@@ -29,28 +30,36 @@ public class MicrosoftIdentityOptions : OpenIdConnectOptions
...
+        /// in containers (for the same reasons).
         /// If you don't specify the PostLogoutRedirectUri, it will be computed by ASP.NET Core using the SignedOutCallbackPath.
         /// </summary>
         public string PostLogoutRedirectUri { get; set; }
 
+        /// <summary>
+        /// When set to true, forces the <see cref="OpenIdConnectMessage.RedirectUri"/> and the <see cref="OpenIdConnectMessage.PostLogoutRedirectUri"/> to use the HTTPS scheme.
+        /// This behavior can be desired, for instance, when you use a reverse proxy that transforms HTTPS
+        /// URLs (external world) to HTTP URLs (inside the protected area). This can also be useful for web apps running
+        /// in containers (for the same reasons).
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web/MicrosoftIdentityOptions.cs:59]:
```suggestion
        /// in containers (for the same reasons), for example when deploying your web app to
        /// Azure App Services in Linux containers.
```

File: tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs [hunk 2]
```diff
@@ -341,6 +341,45 @@ public void AddWebAppCallsProtectedWebApi_NoScopes()
             Assert.Contains(OidcConstants.ScopeProfile, oidcOptions.Scope);
         }
 
+        [Theory]
+        [InlineData(true, "http://localhost:123", "https://localhost:123")]
+        [InlineData(false, "http://localhost:123", "http://localhost:123")]
```
Comment (jmprieur) [inline tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs:346]: Shouldn't there a case with an actual URL starting in https ?

Comment (pmaytak) [reply tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs:347]: Updated.

**Comment by ABAG603** [comment]
Hi. @jmprieur @jennyf19 

Did this change make into any of the releases? I'm currently using version 1.25.1 and not able to find ForceHttpsRedirectUris property.

Thanks

**Pull request merged**